src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java

```diff
@@ -293,8 +297,14 @@ private void handleClientUpdate(RoutingContext rc) {
             final Site site = RequestUtil.getSite(rc, "site_id", this.siteProvider);
             if (site == null) return;
 
+            final int serviceId = this.getServiceId(rc);
+
             existingClient.withSiteId(site.getId());
 
+            if (serviceId != 0) {
```

Comment:
Will there be a case where we do want to convert a client back to 0 service ID? e.g. accidental update

Reply:
Good point. Updated